docs/docs/user/getting-started/working-with-kpops.md

```diff
@@ -0,0 +1,187 @@
+# Working with KPOps
+
+## Components
+
+This section explains the different components of KPOps, their usage and configuration.
+
+### KubernetesApp
+
+#### Usage
+
+Can be used to deploy any app in Kubernetes using helm. An example is a Redis database.
+
+#### Configuration
+
+??? "Title (`file.yaml`)"
+    Description
+
+    ```yaml
+    config:
+    ```
+
+### StreamsApp
+
+#### Usage
+
+Configures a [streams bootstrap app](https://github.com/bakdata/streams-bootstrap) which in turns configures a [Kafka Streams app](https://kafka.apache.org/documentation/streams/)
+
+#### Configuration
+
+??? "Title (`file.yaml`)"
+    Description
+
+    ```yaml
+    config:
+    ```
+
+### Producer
+
+#### Usage
+
+Publish (write) events to a Kafka cluster.
+
+#### Configuration
+
+??? "Title (`file.yaml`)"
+    Description
+
+    ```yaml
+    config:
+    ```
+
+### KafkaSinkConnector
+
+#### Usage
+
+Let other systems pull data from Apache Kafka.
+
+#### Configuration
+
+??? "Title (`file.yaml`)"
+    Description
+
+    ```yaml
+    config:
+    ```
+
+### KafkaSourceConnector
+
+#### Usage
+
+Let other systems push data to Apache Kafka.
+
+#### Configuration
+
+??? "Title (`file.yaml`)"
+    Description
+
+    ```yaml
+    config:
+    ```
+
+## Commands
+
+This section will cover KPOps commands and how they differ depending on the component type.
+
+These commands are
+- deploy
+- destroy
+- reset
+- clean
+
+`KPOps` can apply the command only to a subset of steps by passing `--steps`.
+The parameter takes a comma-separated string of the component's names.
+
+You need to specify either `--dry-run` or `--execute` with each of these commands.
+
+### Deploy
```

Comment:
I would rather do it the other way around. List the components and withing each component describe what deploy/destroy/... do. That way the docs for a component are in one place

Reply:
I thought about that, but that way there would be repetition for example with `destroy`. Maybe you are right though, the commands are already covered in [CLI usage](https://bakdata.github.io/kpops/0.10/user/references/cli-commands/), so it doesn't make sense to cover them again like this here.

Reply:
@philipp94831 last time we discussed this internally, we came to the same conclusion. 
Perhaps it wasn't clear enough from the linked issue.

Reply:
@sujuka99 please restructure so that the methods are described for each component